src/components/footer.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { FilterOptions } from '../types/enums';
+
+type Props = {
+  onSetfilterType: (value: FilterOptions) => void;
+  handleDeleteCompleted: () => void;
+  completedLength: number;
+};
+
+const Footer: React.FC<Props> = ({
+  onSetfilterType,
+  handleDeleteCompleted,
+  completedLength,
+}) => {
+  const [selectedType, setSelectedType] = useState(FilterOptions.All);
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        3 items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={cn('filter__link', {
+            selected: selectedType === FilterOptions.All,
+          })}
+          data-cy="FilterLinkAll"
+          onClick={() => {
+            onSetfilterType(FilterOptions.All);
+            setSelectedType(FilterOptions.All);
+          }}
```

Comment:
why you make some thing twice?
what is the different
why yu can't use one state?
DRY